src/Microsoft.Diagnostics.DotNetMonitor/Microsoft.Diagnostics.DotNetMonitor.csproj

```diff
@@ -0,0 +1,63 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <!-- The intent of this project is to product a NuGet package with self-contained,
+       trimmed deployments of the dotnet-monitor tool. The corresponding tool project
+       cannot publish self-contained because that is explicitly forbidden by projects
+       that pack as .NET tools. -->
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```

Comment:
I take it unification into one project with multiple targets isn't really a viable option because the tool will take everything instead of a specific target?

Reply:
The underlying issue is that any project that is packed as a tool cannot do self-contained publishing.

For this project, I can have it produce netcoreapp2.1 and netcoreapp3.1 and only publish self-contained what I really want to include in the NuGet package that is sent over to dotnet-docker. So this project could be the only one that produces dotnet-monitor.

However, that means I would need to modify the existing dotnet-monitor project to just package the output of this project (that would entail removing what is automatically included in the NuGet package and then adding the build output of this project; the latter is similar to the custom target that I added to the end of this project).

I can explore this option and see how complicated it becomes. If it's acceptable, then I can fold Microsoft.Diagnostics.DotNetMonitor.Core into Microsoft.Diagnostics.DotNetMonitor.